lib/sequent/cli/sequent8_migration.rb

```diff
@@ -0,0 +1,180 @@
+          commit the results.
+
+          Step 4: Check the results and COMMIT or ROLLBACK the result. If you COMMIT,
+          you must perform a VACUUM ANALYZE to ensure PostgreSQL can efficiently query
+          the new tables
+
+          Step 5: Now you can deploy your Sequent 8 based application and start it again.
+
+        EOS
+      end
+
+      def copy_schema_files
+        prompt.say <<~EOS
+
+          Step 1. First a copy of the Sequent 8 database schema and migration scripts are
+          added to your db/ directory.
+        EOS
+        prompt.warn <<~EOS
+
+          WARNING: this may overwrite your existing scripts, please use your version control system to commit or abort any of the changes!
+        EOS
+
+        abort_if_no('Do you which to continue?')
+
+        FileUtils.copy_entry("#{sequent_gem_dir}/db", 'db')
+
+        prompt.warn <<~EOS
+
+          WARNING: The schema files have been copied, please verify and adjust the contents before committing and continuing.
+        EOS
+      end
+
+      def stop_application
+        prompt.say <<~EOS
+
+          Step 2. Please shut down your existing application.
+        EOS
+
+        abort_if_no(<<~EOS)
+          Only proceed once your application is stopped. Is your application stopped and do you want to continue?
+        EOS
+      end
+
+      def migrate_data
+        prompt.say <<~EOS
+
+          Step 3. Open a `psql` connection to the database you wish to migrate.
+        EOS
+        prompt.warn <<~EOS
+
+          It is highly recommended to test this on a copy of your production database first!
+        EOS
+
+        prompt.say <<~EOS
+
+          Depending on the size of your database the migration can take a long time. Open the `psql` connection from a screen session if needed.
+          If you run this from a screen session from another server you will need to copy all needed sql files to that server.
+
+          ```
+          psql -U myapp_user myapp_db
+          ```
+        EOS
+
+        prompt.ask('Press <enter> to read the next instructions once you connected to the database...')
+
+        prompt.say <<~EOS
+
+          Run the database migration. This doesn't commit anything yet so you can check the results first.
+
+          ```
+          psql> \\i db/sequent_8_migration.sql
+          ```
+        EOS
+      end
+
+      def commit_or_rollback
+        answer = prompt.yes? 'Did the migration succeed?'
+        if answer
+          prompt.say <<~EOS
+
+            Step 4. After checking everything went OK, COMMIT and optimize the database:
+
+            ```
+            psql> COMMIT; VACUUM VERBOSE ANALYZE;
+            ```
+          EOS
+        else
+          prompt.say <<~EOS
+
+            Step 4. Rollback the migration:
+
+            ```
+            psql> ROLLBACK;
+            ```
+          EOS
+        end
+        answer
+      end
+
+      def sequent_gem_dir = Gem::Specification.find_by_name('sequent').gem_dir
```

Comment:
This breaks when running directly from the sequent source directory. Is there a way to make a gem work without having to install it first? Or another easy way to test this code without having to install every time?

Reply:
Yes, you could probably do something like:

`some_file.rb`:
```ruby
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "sequent", git: 'https://github.com/zilverline/sequent'
end

require 'gli'
require 'tty-prompt'
require './lib/version'
require './lib/sequent/cli/app'

exit_code = Sequent::Cli::App.run(ARGV)
exit(exit_code)
```